components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -797,6 +797,12 @@ private void addAuthorizedInternalScopes(OAuthTokenReqMessageContext tokReqMsgCt
                                              String[] authorizedInternalScopes) {
 
         String[] scopes = tokReqMsgCtx.getScope();
+        if (ArrayUtils.isEmpty(scopes)) {
+            scopes = new String[0];
```

Comment:
Are we checking this empty scope array from any place and do the logic based on that?